Notes:
- Compare CA median income and/or home price and/or HIR and/or house share of inc to other states (if it's at the extreme, then it's not significant if many high mig states are higher or lower)
- Identify counties where there's a big diff b/w mig into vs. out of
- Compare incomes & home prices of Net In vs. Net Out counties

In [1]:
import pandas as pd
import numpy as np

In [2]:
%run chart_theme.py

In [2]:
fin = pd.read_csv('county_rent_mort_inc_units_5yr.csv')

In [3]:
fin = fin[['FIPS', 'County', 'Population', 'Med_Mort', 'Med_Rent', 'Med_Mort_Inc', 'Med_Rent_Inc', 'Mort_Share_Inc', 'Rent_Share_Inc']][fin.Year == 2015]

In [5]:
fin.head(1)

FIPS                   County  Population  Med_Mort  Med_Rent  \
21839  1001  Autauga County, Alabama       55036     853.0     932.0   

       Med_Mort_Inc  Med_Rent_Inc  Mort_Share_Inc  Rent_Share_Inc  
21839        5253.0        3053.0            16.2            30.5

In [6]:
fin[fin.County.str.contains('Cobb')]

FIPS                County  Population  Med_Mort  Med_Rent  \
22254  13067  Cobb County, Georgia      739072    1294.0    1102.0   

       Med_Mort_Inc  Med_Rent_Inc  Mort_Share_Inc  Rent_Share_Inc  
22254        7658.0        3799.0            16.9            29.0

In [7]:
fin[fin.County.str.contains('Fulton')]

FIPS                       County  Population  Med_Mort  Med_Rent  \
21970   5049      Fulton County, Arkansas       12092     597.0     504.0   
22281  13121       Fulton County, Georgia     1010420    1458.0    1089.0   
22455  17057      Fulton County, Illinois       35733     668.0     631.0   
22553  18049       Fulton County, Indiana       20296     675.0     639.0   
22861  21075      Fulton County, Kentucky        6259     478.0     501.0   
23670  36035      Fulton County, New York       53955     791.0     734.0   
23892  39051          Fulton County, Ohio       42296     886.0     683.0   
24096  42057  Fulton County, Pennsylvania       14631     667.0     663.0   

       Med_Mort_Inc  Med_Rent_Inc  Mort_Share_Inc  Rent_Share_Inc  
21970        3683.0        1318.0            16.2            38.2  
22281        8032.0        3314.0            18.2            32.9  
22455        4690.0        2042.0            14.2            30.9  
22553        4332.0        2614.0            15.6            24.4  
22861        3229.0        1555.0            14.8            32.2  
23670        4870.0        2412.0            16.2            30.4  
23892        5619.0        2523.0            15.8            27.1  
24096        4698.0        2693.0            14.2            24.6

---

### CA Counties Mig (5-yr)

In [6]:
# Files created in ca_counties_mig_5yr_0917 file
file1 = pd.read_csv('ca_counties_mig_5yr_0917_adj.csv')
file2 = pd.read_csv('ca_counties_mig_5yr_0917.csv')

In [7]:
file1['Address'] = file1.County2Name + ', ' + file1.State2Name

In [8]:
# Excludes international migration
noint1 = file1[(file1.County2FIPS > 0) & (file1.County2FIPS < 6000) | (file1.County2FIPS > 7000)]
noint2 = file2[(file2.County2FIPS > 0) & (file2.County2FIPS < 6000) | (file2.County2FIPS > 7000)]

In [9]:
for year in noint1.Year.unique():
    df = noint1[noint1.Year == year]
    print(year, df.MovedIn.sum(), df.MovedOut.sum(), df.MovedIn.sum() - df.MovedOut.sum())

2009 485529 670121 -184592
2010 474317 629820 -155503
2011 465854 591974 -126120
2012 467335 568282 -100947
2013 473266 567557 -94291
2014 487074 575064 -87990
2015 497313 588108 -90795
2016 503849 603516 -99667
2017 509797 620792 -110995


In [10]:
for year in noint2.Year.unique():
    df = noint2[noint2.Year == year]
    print(year, df.MovedIn.sum(), df.MovedOut.sum(), df.MovedIn.sum() - df.MovedOut.sum())

2009 582440 750632 -168192
2010 568575 711169 -142594
2011 555046 672174 -117128
2012 556382 648139 -91757
2013 563692 649837 -86145
2014 583127 658151 -75024
2015 597946 669923 -71977
2016 611126 685487 -74361
2017 620351 704615 -84264


In [12]:
noint1.head()

County1Name  County1FIPS State2Name     County2Name  County2FIPS  \
9   Alameda County         6001    Alabama  Autauga County         1001   
10  Alameda County         6001    Alabama   Blount County         1009   
11  Alameda County         6001    Alabama  Calhoun County         1015   
12  Alameda County         6001    Alabama     Dale County         1045   
13  Alameda County         6001    Alabama   Dallas County         1047   

    County1Population  MovedIn  MovedOut  Net_Dom  NonMovers  \
9             1437189        0       212     -212    1197957   
10            1437189        0        30      -30    1197957   
11            1437189        0        11      -11    1197957   
12            1437189        0         9       -9    1197957   
13            1437189        0        22      -22    1197957   

    Within_Same_County  From_Diff_County_Same_State  From_Diff_State  \
9               143538                        58630            21729   
10              143538                        58630            21729   
11              143538                        58630            21729   
12              143538                        58630            21729   
13              143538                        58630            21729   

    From_Abroad  To_Diff_County_Same_State  To_Diff_State  To_PR  State  Year  \
9         15335                      62492          25040      0      6  2009   
10        15335                      62492          25040      0      6  2009   
11        15335                      62492          25040      0      6  2009   
12        15335                      62492          25040      0      6  2009   
13        15335                      62492          25040      0      6  2009   

                    Address  
9   Autauga County, Alabama  
10   Blount County, Alabama  
11  Calhoun County, Alabama  
12     Dale County, Alabama  
13   Dallas County, Alabama

In [13]:
noint2.head()

County1Name  County1FIPS State2Name     County2Name  County2FIPS  \
9   Alameda County         6001    Alabama  Autauga County         1001   
10  Alameda County         6001    Alabama   Blount County         1009   
11  Alameda County         6001    Alabama  Calhoun County         1015   
12  Alameda County         6001    Alabama     Dale County         1045   
13  Alameda County         6001    Alabama   Dallas County         1047   

    County1Population  MovedIn  MovedOut  Net_Dom  NonMovers  \
9             1437189        0       212     -212    1197957   
10            1437189        0        30      -30    1197957   
11            1437189        0        11      -11    1197957   
12            1437189        0         9       -9    1197957   
13            1437189        0        22      -22    1197957   

    Within_Same_County  From_Diff_County_Same_State  From_Diff_State  \
9               143538                        58630            21729   
10              143538                        58630            21729   
11              143538                        58630            21729   
12              143538                        58630            21729   
13              143538                        58630            21729   

    From_Abroad  To_Diff_County_Same_State  To_Diff_State  To_PR  State  Year  
9         15335                      62492          25040      0      6  2009  
10        15335                      62492          25040      0      6  2009  
11        15335                      62492          25040      0      6  2009  
12        15335                      62492          25040      0      6  2009  
13        15335                      62492          25040      0      6  2009

In [5]:
la17 = file1[(file1.County1FIPS == 6037) & (file1.County2FIPS < 60000) & (file1.Year == 2017)]

In [11]:
ca17 = file1[((file1.County1FIPS > 6000) & (file1.County1FIPS < 7000)) & (file1.County2FIPS < 60000) & (file1.State2Name != 'California') & (file1.Year == 2017)].sort_values('MovedOut', ascending=False)

In [12]:
ca17.head(1)

County1Name  County1FIPS State2Name   County2Name  County2FIPS  \
143073  Los Angeles County         6037     Nevada  Clark County        32003   

        County1Population  MovedIn  MovedOut  Net_Dom  NonMovers  \
143073            9988370     4724     14191    -9467    8849567   

        Within_Same_County  From_Diff_County_Same_State  From_Diff_State  \
143073              851215                       110785           100322   

        From_Abroad  To_Diff_County_Same_State  To_Diff_State  To_PR  State  \
143073        76481                     180991         135437    370      6   

        Year               Address  
143073  2017  Clark County, Nevada

In [13]:
len(ca17.County2FIPS.unique())

2302

CA Out-Migration:

In [15]:
ca17.groupby('County2FIPS')['MovedOut'].sum()

County2FIPS
0          0
1001     150
1003     240
1009      11
1015     149
        ... 
56037    325
56039    435
56041     40
56043     12
56045      2
Name: MovedOut, Length: 2302, dtype: int64

#### Top 50 Out & In Migration Counties:

In [20]:
file2.Year.sort_values().unique()

array([2009, 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017], dtype=int64)

In [82]:
temp_ca_outmig = ca17.groupby('County2FIPS').MovedOut.sum()  # Calculating total CA outmig figures for each non-CA county
ca_outmig = pd.DataFrame({'FIPS':temp_ca_outmig.index, 'Outmig':temp_ca_outmig.values})[1:]  # Removing first row (Int'l migration)
counties = []
states = []
# Adding County,State col (for DataWrapper map coordinates)
for fips in ca_outmig.FIPS.unique():
    counties.append(ca17.County2Name[ca17.County2FIPS == fips].iloc[0])
    states.append(ca17.State2Name[ca17.County2FIPS == fips].iloc[0])
    
    ca_outmig['County_Name'] = counties
    ca_outmig['State_Name'] = states
    ca_outmig['County_State'] = ca_outmig.County_Name + ', ' + ca_outmig.State_Name
    ca_outmig = ca_outmig.drop(['County_Name', 'State_Name'], axis=1)

In [83]:
temp_ca_inmig = ca17.groupby('County2FIPS').MovedIn.sum()  # Calculating total CA outmig figures for each non-CA county
ca_inmig = pd.DataFrame({'FIPS':temp_ca_inmig.index, 'Inmig':temp_ca_inmig.values})[1:]  # Removing first row (Int'l migration)
counties = []
states = []
for fips in ca_inmig.FIPS.unique():
    counties.append(ca17.County2Name[ca17.County2FIPS == fips].iloc[0])
    states.append(ca17.State2Name[ca17.County2FIPS == fips].iloc[0])
    
ca_inmig['County_Name'] = counties
ca_inmig['State_Name'] = states
ca_inmig['County_State'] = ca_inmig.County_Name + ', ' + ca_inmig.State_Name
ca_inmig = ca_inmig.drop(['County_Name', 'State_Name'], axis=1)

---

In [ ]:
census5yr = pd.read_csv('')
for year in census5yr.Year.unique():
    df = census5yr[census5yr.Year == year]
    ca_out_in = pd.DataFrame()
    counter = 0
    mig_types = ['MovedOut', 'MovedIn']
    for mig in mig_types:
        series = df.groupby('County2FIPS')[mig].sum()  # Calculating total CA outmig figures for each non-CA county
        ca_mig = pd.DataFrame({'FIPS':series.index, mig:series.values})[1:]  # Removing first row (Int'l migration)
        counties = []
        states = []
        # Adding County,State col (for DataWrapper map coordinates)
        for fips in ca_mig.FIPS.unique():
            counties.append(df.County2Name[df.County2FIPS == fips].iloc[0])
            states.append(df.State2Name[df.County2FIPS == fips].iloc[0])

        ca_mig['County_Name'] = counties
        ca_mig['State_Name'] = states
        ca_mig['County_State'] = ca_mig.County_Name + ', ' + ca_mig.State_Name
        ca_mig = ca_mig.drop(['County_Name', 'State_Name'], axis=1)

        if counter == 0:
            ca_out_in = ca_mig.copy()
        elif counter == 1:
            ca_out_in = ca_out_in.merge(ca_mig, on=['FIPS', 'County_State'])
            ca_out_in['Net_Mig'] = ca_out_in.Inmig - ca_out_in.Outmig
            ca_out_in = ca_out_in.sort_values('Net_Mig')
        counter += 1

    top50_out_in = pd.concat([ca_out_in.iloc[:50], ca_out_in.iloc[-50:]])
    top50_out_in['Mig_Abs'] = top50_out_in.Net_Mig.abs()
    top50_out_in['Type'] = ['Net Out']*50 + ['Net In']*50
    top50_out_in['More'] = [c.split(',')[0] for c in top50_out_in.County_State[:50].values] + list(np.full(50, 'California'))
    top50_out_in['Year'] = np.full(len(top50_out_in), year)
    top50_out_in.to_csv('Census_5yr_CA_Netmig_top50_in_out_counties_' + str(year) + '.csv', index=False)

In [88]:
# Combining inmig & outmig data
ca_out_in = ca_outmig.merge(ca_inmig, on=['FIPS', 'County_State'])
ca_out_in['Net_Mig'] = ca_out_in.Inmig - ca_out_in.Outmig
ca_out_in = ca_out_in.sort_values('Net_Mig')

In [100]:
top50_out_in = pd.concat([ca_out_in.iloc[:50], ca_out_in.iloc[-50:]])
top50_out_in['Mig_Abs'] = top50_out_in.Net_Mig.abs()
top50_out_in['Type'] = ['Net Out']*50 + ['Net In']*50
top50_out_in['More'] = [c.split(',')[0] for c in top50_out_in.County_State[:50].values] + list(np.full(50, 'California'))
top50_out_in.to_csv('Census_5yr_CA_Netmig_top50_in_out_counties_1317.csv', index=False)

In [101]:
ca_out_in

FIPS  Outmig                       County_State  Inmig  Net_Mig
1231  32003   33834               Clark County, Nevada  17993   -15841
72     4013   31213           Maricopa County, Arizona  18187   -13026
1632  41051    8293           Multnomah County, Oregon   4533    -3760
1883  48113    7585               Dallas County, Texas   3967    -3618
362   16001    5593                  Ada County, Idaho   1996    -3597
...     ...     ...                                ...    ...      ...
1690  42101    1552  Philadelphia County, Pennsylvania   3102     1550
358   15003    8078            Honolulu County, Hawaii   9998     1920
1333  36061    8708          New York County, New York  10697     1989
1326  36047    4000             Kings County, New York   7772     3772
413   17031    7789              Cook County, Illinois  13718     5929

[2301 rows x 5 columns]

In [102]:
ca_mig_fin = ca_out_in.merge(fin, on='FIPS')
#combined_all['MovedIn_Per'] = combined_all.MovedIn / combined_all.Population * 100
#combined_all['MovedOut_Per'] = combined_all.MovedOut / combined_all.Population * 100

In [103]:
ca_mig_fin

FIPS  Outmig                       County_State  Inmig  Net_Mig  \
0     32003   33834               Clark County, Nevada  17993   -15841   
1      4013   31213           Maricopa County, Arizona  18187   -13026   
2     41051    8293           Multnomah County, Oregon   4533    -3760   
3     48113    7585               Dallas County, Texas   3967    -3618   
4     16001    5593                  Ada County, Idaho   1996    -3597   
...     ...     ...                                ...    ...      ...   
2289  42101    1552  Philadelphia County, Pennsylvania   3102     1550   
2290  15003    8078            Honolulu County, Hawaii   9998     1920   
2291  36061    8708          New York County, New York  10697     1989   
2292  36047    4000             Kings County, New York   7772     3772   
2293  17031    7789              Cook County, Illinois  13718     5929   

                                 County  Population  Med_Mort  Med_Rent  \
0                  Clark County, Nevada     2112436    1159.0    1048.0   
1              Maricopa County, Arizona     4155501    1144.0    1033.0   
2              Multnomah County, Oregon      788459    1531.0    1094.0   
3                  Dallas County, Texas     2552213    1161.0     984.0   
4                     Ada County, Idaho      435117    1111.0     910.0   
...                                 ...         ...       ...       ...   
2289  Philadelphia County, Pennsylvania     1569657     890.0     970.0   
2290            Honolulu County, Hawaii      990060    1873.0    1653.0   
2291          New York County, New York     1653877    1889.0    1615.0   
2292             Kings County, New York     2635121    1844.0    1314.0   
2293              Cook County, Illinois     5238541    1446.0    1044.0   

      Med_Mort_Inc  Med_Rent_Inc  Mort_Share_Inc  Rent_Share_Inc  
0           5901.0        3401.0            19.6            30.8  
1           6220.0        3395.0            18.4            30.4  
2           7110.0        3224.0            21.5            33.9  
3           6109.0        3344.0            19.0            29.4  
4           6254.0        3024.0            17.8            30.1  
...            ...           ...             ...             ...  
2289        4489.0        2382.0            19.8            40.7  
2290        8730.0        4751.0            21.5            34.8  
2291       12394.0        5348.0            15.2            30.2  
2292        7378.0        3449.0            25.0            38.1  
2293        6724.0        3164.0            21.5            33.0  

[2294 rows x 13 columns]

In [105]:
ca_mig_fin[ca_mig_fin.County_State.isin(['Fulton County, Georgia', 'Cobb County, Georgia'])]

FIPS  Outmig            County_State  Inmig  Net_Mig  \
45    13067    1734    Cobb County, Georgia    888     -846   
2277  13121    1671  Fulton County, Georgia   2769     1098   

                      County  Population  Med_Mort  Med_Rent  Med_Mort_Inc  \
45      Cobb County, Georgia      739072    1294.0    1102.0        7658.0   
2277  Fulton County, Georgia     1010420    1458.0    1089.0        8032.0   

      Med_Rent_Inc  Mort_Share_Inc  Rent_Share_Inc  
45          3799.0            16.9            29.0  
2277        3314.0            18.2            32.9

*Use IPUMS data to compare migrants to/from Cobb vs. Fulton?

---

In [73]:
temp_ca_outmig = ca17.groupby('County2FIPS').MovedOut.sum()
ca_outmig = pd.DataFrame({'FIPS':temp_ca_outmig.index, 'Mig':temp_ca_outmig.values})[1:]  # Removing first row (Int'l migration)
counties = []
states = []
for fips in ca_outmig.FIPS.unique():
    counties.append(ca17.County2Name[ca17.County2FIPS == fips].iloc[0])
    states.append(ca17.State2Name[ca17.County2FIPS == fips].iloc[0])
    
ca_outmig['County_Name'] = counties
ca_outmig['State_Name'] = states
ca_outmig['County_State'] = ca_outmig.County_Name + ', ' + ca_outmig.State_Name
ca_outmig['Mig_Type'] = np.full(len(ca_outmig), 'Out')

In [74]:
ca_outmig.head(3)

FIPS  Mig     County_Name State_Name             County_State Mig_Type
1  1001  150  Autauga County    Alabama  Autauga County, Alabama      Out
2  1003  240  Baldwin County    Alabama  Baldwin County, Alabama      Out
3  1009   11   Blount County    Alabama   Blount County, Alabama      Out

In [80]:
len(ca_outmig)

2301

In [72]:
temp_ca_inmig = ca17.groupby('County2FIPS').MovedIn.sum()
ca_inmig = pd.DataFrame({'FIPS':temp_ca_inmig.index, 'Mig':temp_ca_inmig.values})[1:]  # Removing first row (Int'l migration)
counties = []
states = []
for fips in ca_inmig.FIPS.unique():
    counties.append(ca17.County2Name[ca17.County2FIPS == fips].iloc[0])
    states.append(ca17.State2Name[ca17.County2FIPS == fips].iloc[0])
    
ca_inmig['County_Name'] = counties
ca_inmig['State_Name'] = states
ca_inmig['County_State'] = ca_inmig.County_Name + ', ' + ca_inmig.State_Name
ca_inmig['Mig_Type'] = np.full(len(ca_inmig), 'In')

In [75]:
ca_inmig.head(3)

FIPS  Mig     County_Name State_Name             County_State Mig_Type
1  1001    9  Autauga County    Alabama  Autauga County, Alabama       In
2  1003  176  Baldwin County    Alabama  Baldwin County, Alabama       In
3  1009  154   Blount County    Alabama   Blount County, Alabama       In

In [81]:
len(ca_inmig)

2301

In [77]:
top50out = ca_outmig.sort_values('Mig', ascending=False).iloc[:50]
top50in = ca_inmig.sort_values('Mig', ascending=False).iloc[:50]
top50_in_out = pd.concat([top50in, top50out])

In [78]:
top50_in_out

FIPS    Mig       County_Name      State_Name  \
73     4013  18187   Maricopa County         Arizona   
1232  32003  17993      Clark County          Nevada   
2163  53033  14410       King County      Washington   
414   17031  13718       Cook County        Illinois   
1334  36061  10697   New York County        New York   
...     ...    ...               ...             ...   
439   17097   2562       Lake County        Illinois   
1617  41017   2408  Deschutes County          Oregon   
1611  41005   2369  Clackamas County          Oregon   
1435  37183   2361       Wake County  North Carolina   
174    8069   2297    Larimer County        Colorado   

                     County_State Mig_Type  
73       Maricopa County, Arizona       In  
1232         Clark County, Nevada       In  
2163      King County, Washington       In  
414         Cook County, Illinois       In  
1334    New York County, New York       In  
...                           ...      ...  
439         Lake County, Illinois      Out  
1617     Deschutes County, Oregon      Out  
1611     Clackamas County, Oregon      Out  
1435  Wake County, North Carolina      Out  
174      Larimer County, Colorado      Out  

[100 rows x 6 columns]

In [55]:
top50_in_out.to_csv('Census_5yr_CA_Outmig_counties_1317.csv', index=False)

In [79]:
top50_in_out.to_csv('Census_5yr_CA_Outmig_top50_in_out_counties_1317.csv', index=False)

In [42]:
ca17.groupby('County2Name').agg({'County2FIPS':'max','MovedOut':'sum'}).sort_values('County2FIPS')

County2FIPS  MovedOut
County2Name                             
0                            0         0
Autauga County            1001       150
Colbert County            1033        48
Covington County          1039        35
Cullman County            1043        88
...                        ...       ...
Sweetwater County        56037       325
Teton County             56039       486
Uinta County             56041        40
Washakie County          56043        12
Weston County            56045         2

[1460 rows x 2 columns]

In [32]:
ca17.groupby('County2Name').sum().sort_values('MovedOut', ascending=False)

MovedOut  County2FIPS
County2Name                             
Clark County          37231        55019
Maricopa County       31213         4013
King County           16505        53033
Washoe County          9377        32031
Washington County      8910        55131
Harris County          8816        48201
New York County        8708        36061
Multnomah County       8293        41051
Honolulu County        8078        15003
Dallas County          8029        48113
Cook County            7793        27031
Douglas County         6912        55031
Pima County            6844         4019
El Paso County         6449        48141
Tarrant County         6186        48439

In [32]:
ca17.groupby('County2Name').agg({'MovedOut':'sum', 'County2FIPS':'max'}).sort_values('MovedOut', ascending=False).head(15)

MovedOut  County2FIPS
County2Name                             
Clark County          37231        55019
Maricopa County       31213         4013
King County           16505        53033
Washoe County          9377        32031
Washington County      8910        55131
Harris County          8816        48201
New York County        8708        36061
Multnomah County       8293        41051
Honolulu County        8078        15003
Dallas County          8029        48113
Cook County            7793        27031
Douglas County         6912        55031
Pima County            6844         4019
El Paso County         6449        48141
Tarrant County         6186        48439

In [150]:
# Moved Out of LA county
total = la17.sort_values('MovedOut', ascending=False)
in_ca = la17[la17.State2Name == 'California'].sort_values('MovedOut', ascending=False)
out_ca = la17[(la17.State2Name != 'California') & (la17.County2FIPS > 0)].sort_values('MovedOut', ascending=False)

In [151]:
out_ca['MovedIn_Str'] = out_ca.MovedIn.apply(lambda x:'{:,}'.format(x))
out_ca['MovedOut_Str'] = out_ca.MovedOut.apply(lambda x:'{:,}'.format(x))
out_ca['MovedNet_Str'] = out_ca.Net_Dom.apply(lambda x:'{:,}'.format(x))

In [152]:
out_ca.head(1)

County1Name  County1FIPS State2Name   County2Name  County2FIPS  \
143073  Los Angeles County         6037     Nevada  Clark County        32003   

        County1Population  MovedIn  MovedOut  Net_Dom  NonMovers  ...  \
143073            9988370     4724     14191    -9467    8849567  ...   

        From_Diff_State  From_Abroad  To_Diff_County_Same_State  \
143073           100322        76481                     180991   

        To_Diff_State  To_PR  Year               Address  MovedIn_Str  \
143073         135437    370  2017  Clark County, Nevada        4,724   

       MovedOut_Str MovedNet_Str  
143073       14,191       -9,467  

[1 rows x 22 columns]

In [153]:
out_ca.sort_values('MovedIn', ascending=False).head(50).to_csv('CA_top_50_inmig_1317.csv', index=False)
out_ca.sort_values('MovedOut', ascending=False).head(50).to_csv('CA_top_50_outmig_1317.csv', index=False)

---

#### Top 50 Out & In Migration Counties:

In [154]:
outmig = out_ca.sort_values('Net_Dom').iloc[-50:]
outmig['Type'] = list(np.full(len(outmig),'Net In'))
outmig['More'] = list(np.full(len(outmig), 'LA County'))
inmig = out_ca.sort_values('Net_Dom').iloc[:50]
inmig['Type'] = list(np.full(len(inmig),'Net Out'))
inmig['More'] = inmig.County2Name.values
inmig.Net_Dom = inmig.Net_Dom.apply(lambda x:abs(x))
outmig_inmig = pd.concat([outmig, inmig])

In [ ]:
outmig_inmig.to_csv('LA_top_50_net_inmig_outmig_1317.csv', index=False)

In [157]:
for fips in outmig_inmig.County2FIPS.unique():
    if fips not in fin.FIPS.values:
        print(fips)

In [159]:
fips_codes = outmig_inmig.County2FIPS.unique()
financials = fin[fin.FIPS.isin(fips_codes)]

In [161]:
combined = outmig_inmig.merge(financials, left_on='County2FIPS', right_on='FIPS')

In [166]:
combined.Med_Mort[combined.Type == 'Net Out'].median()

1231.0

In [167]:
combined.Med_Rent[combined.Type == 'Net Out'].median()

1022.0

In [168]:
combined.Med_Mort[combined.Type == 'Net In'].median()

1361.5

In [169]:
combined.Med_Rent[combined.Type == 'Net In'].median()

1053.0

In [170]:
combined.Med_Mort_Inc[combined.Type == 'Net Out'].median()

6632.0

In [171]:
combined.Med_Rent_Inc[combined.Type == 'Net Out'].median()

3386.5

In [172]:
combined.Med_Mort_Inc[combined.Type == 'Net In'].median()

7182.0

In [173]:
combined.Med_Rent_Inc[combined.Type == 'Net In'].median()

3173.5

---

#### All Out & In Migration Counties:

In [385]:
outmig_all = out_ca[out_ca.MovedIn != out_ca.MovedOut]  # Only interested in comparing counties where one is higher than the other

In [386]:
for fips in outmig_all.County2FIPS.unique():
    if fips not in fin.FIPS.values:
        print(fips)

35047
37189


In [387]:
len(outmig_all)

1235

In [388]:
outmig_all = outmig_all[~outmig_all.County2FIPS.isin([35047, 37189])]

In [389]:
len(outmig_all)

1233

In [390]:
all_fips_codes = outmig_all.County2FIPS.unique()
all_financials = fin[fin.FIPS.isin(all_fips_codes)]

In [407]:
combined_all = outmig_all.merge(all_financials, left_on='County2FIPS', right_on='FIPS')
combined_all['MovedIn_Per'] = combined_all.MovedIn / combined_all.Population * 100
combined_all['MovedOut_Per'] = combined_all.MovedOut / combined_all.Population * 100

In [408]:
combined_all.head(1)

County1Name  County1FIPS State2Name   County2Name  County2FIPS  \
0  Los Angeles County         6037     Nevada  Clark County        32003   

   County1Population  MovedIn  MovedOut  Net_Dom  NonMovers  ...   FIPS  \
0            9988370     4724     14191    -9467    8849567  ...  32003   

   Population  Med_Mort  Med_Rent  Med_Mort_Inc  Med_Rent_Inc  Mort_Share_Inc  \
0     2112436    1159.0    1048.0        5901.0        3401.0            19.6   

   Rent_Share_Inc MovedIn_Per MovedOut_Per  
0            30.8    0.223628     0.671784  

[1 rows x 32 columns]

In [393]:
combined_all.Med_Mort.min()

240.0

In [394]:
combined_all.Med_Mort.max()

2539.0

In [395]:
combined_all.Med_Rent.min()

330.0

In [396]:
combined_all.Med_Rent.max()

1895.0

---

In [409]:
comb_in = combined_all[combined_all.MovedIn > combined_all.MovedOut]
comb_out = combined_all[combined_all.MovedOut > combined_all.MovedIn]

---

In [410]:
comb_in.columns

Index(['County1Name', 'County1FIPS', 'State2Name', 'County2Name',
       'County2FIPS', 'County1Population', 'MovedIn', 'MovedOut', 'Net_Dom',
       'NonMovers', 'Within_Same_County', 'From_Diff_County_Same_State',
       'From_Diff_State', 'From_Abroad', 'To_Diff_County_Same_State',
       'To_Diff_State', 'To_PR', 'Year', 'Address', 'MovedIn_Str',
       'MovedOut_Str', 'MovedNet_Str', 'FIPS', 'Population', 'Med_Mort',
       'Med_Rent', 'Med_Mort_Inc', 'Med_Rent_Inc', 'Mort_Share_Inc',
       'Rent_Share_Inc', 'MovedIn_Per', 'MovedOut_Per'],
      dtype='object')

In [419]:
alt.Chart(comb_in[comb_in.Population < 1000000]).mark_circle().encode(x='Population', y='MovedIn_Per')

alt.Chart(...)

In [416]:
comb_in.head()

County1Name  County1FIPS            State2Name  \
3   Los Angeles County         6037              New York   
6   Los Angeles County         6037              Illinois   
12  Los Angeles County         6037              New York   
13  Los Angeles County         6037                Hawaii   
16  Los Angeles County         6037  District of Columbia   

             County2Name  County2FIPS  County1Population  MovedIn  MovedOut  \
3        New York County        36061            9988370     3804      3316   
6            Cook County        17031            9988370     3784      2402   
12          Kings County        36047            9988370     3440      1298   
13       Honolulu County        15003            9988370     1855      1280   
16  District of Columbia        11001            9988370     1275      1175   

    Net_Dom  NonMovers  ...   FIPS  Population  Med_Mort  Med_Rent  \
3       488    8849567  ...  36061     1653877    1889.0    1615.0   
6      1382    8849567  ...  17031     5238541    1446.0    1044.0   
12     2142    8849567  ...  36047     2635121    1844.0    1314.0   
13      575    8849567  ...  15003      990060    1873.0    1653.0   
16      100    8849567  ...  11001      672391    1955.0    1424.0   

    Med_Mort_Inc  Med_Rent_Inc  Mort_Share_Inc  Rent_Share_Inc MovedIn_Per  \
3        12394.0        5348.0            15.2            30.2    0.230005   
6         6724.0        3164.0            21.5            33.0    0.072234   
12        7378.0        3449.0            25.0            38.1    0.130544   
13        8730.0        4751.0            21.5            34.8    0.187362   
16       10213.0        4320.0            19.1            33.0    0.189622   

   MovedOut_Per  
3      0.200499  
6      0.045852  
12     0.049258  
13     0.129285  
16     0.174750  

[5 rows x 32 columns]

In [411]:
len(comb_in)

432

In [415]:
comb_in[(comb_in.MovedIn > 50) & (comb_in.MovedIn_Per > .1)]

County1Name  County1FIPS     State2Name            County2Name  \
3     Los Angeles County         6037       New York        New York County   
21    Los Angeles County         6037  Massachusetts         Suffolk County   
415   Los Angeles County         6037        Georgia   Chattahoochee County   
565   Los Angeles County         6037     Washington        Okanogan County   
695   Los Angeles County         6037       Virginia         Lexington city   
752   Los Angeles County         6037         Alaska  Kodiak Island Borough   
992   Los Angeles County         6037           Iowa       Jefferson County   
1038  Los Angeles County         6037        Wyoming        Big Horn County   
1051  Los Angeles County         6037        Alabama         Pickens County   
1061  Los Angeles County         6037       Virginia      Martinsville city   
1076  Los Angeles County         6037           Iowa        Crawford County   
1088  Los Angeles County         6037        Indiana          Jasper County   
1089  Los Angeles County         6037       Illinois           Stark County   
1092  Los Angeles County         6037      Wisconsin          Oconto County   
1110  Los Angeles County         6037         Kansas           Trego County   
1115  Los Angeles County         6037         Kansas          Barton County   
1127  Los Angeles County         6037         Oregon            Lake County   
1146  Los Angeles County         6037   North Dakota        Sheridan County   
1148  Los Angeles County         6037   North Dakota        McKenzie County   
1154  Los Angeles County         6037      Tennessee         Stewart County   
1166  Los Angeles County         6037   South Dakota         Bennett County   
1167  Los Angeles County         6037          Texas      Deaf Smith County   
1189  Los Angeles County         6037       Virginia        Accomack County   
1231  Los Angeles County         6037         Alaska         Denali Borough   

      County2FIPS  County1Population  MovedIn  MovedOut  Net_Dom  NonMovers  \
3           36061            9988370     3804      3316      488    8849567   
21          25025            9988370     1667       970      697    8849567   
415         13053            9988370       56        45       11    8849567   
565         53047            9988370      172        25      147    8849567   
695         51678            9988370       19        13        6    8849567   
752          2150            9988370       35        11       24    8849567   
992         19101            9988370      110         2      108    8849567   
1038        56003            9988370       60         0       60    8849567   
1051         1107            9988370       44         0       44    8849567   
1061        51690            9988370       31         0       31    8849567   
1076        19047            9988370       37         0       37    8849567   
1088        18073            9988370      104         0      104    8849567   
1089        17175            9988370       19         0       19    8849567   
1092        55083            9988370       75         0       75    8849567   
1110        20195            9988370       20         0       20    8849567   
1115        20009            9988370      121         0      121    8849567   
1127        41037            9988370      179         0      179    8849567   
1146        38083            9988370       19         0       19    8849567   
1148        38053            9988370       24         0       24    8849567   
1154        47161            9988370       27         0       27    8849567   
1166        46007            9988370        7         0        7    8849567   
1167        48117            9988370       89         0       89    8849567   
1189        51001            9988370       74         0       74    8849567   
1231         2068            9988370       45         0       45    8849567   

      ...   FIPS  Population  Med_Mort  Med_Rent  Med_Mort_In

In [361]:
len(comb_in[comb_in.MovedIn - comb_in.MovedOut > 50])

114

In [379]:
# 10%+ more people moved in than out
len(comb_in[(comb_in.MovedIn / comb_in.MovedOut > 1.2) & (comb_in.MovedIn > 100)])

82

In [378]:
# Twice as many people moved in (small numbers) or 20% more (more than 100 people moved in)
len(comb_in[((comb_in.MovedIn / comb_in.MovedOut > 1.2) & (comb_in.MovedIn > 100)) | ((comb_in.MovedIn / comb_in.MovedOut > 2) & (comb_in.MovedIn > 50))])

127

In [421]:
comb_in[((comb_in.MovedIn / comb_in.MovedOut > 1.2) & (comb_in.MovedIn > 100)) | ((comb_in.MovedIn / comb_in.MovedOut > 2) & (comb_in.MovedIn > 50))].sort_values('MovedIn').iloc[60:90]

County1Name  County1FIPS      State2Name  \
329   Los Angeles County         6037        New York   
237   Los Angeles County         6037        Missouri   
305   Los Angeles County         6037  North Carolina   
220   Los Angeles County         6037   Massachusetts   
272   Los Angeles County         6037   Massachusetts   
264   Los Angeles County         6037        Virginia   
461   Los Angeles County         6037      New Jersey   
314   Los Angeles County         6037            Ohio   
253   Los Angeles County         6037        Illinois   
533   Los Angeles County         6037         Indiana   
259   Los Angeles County         6037        Arkansas   
344   Los Angeles County         6037      New Jersey   
507   Los Angeles County         6037        Kentucky   
565   Los Angeles County         6037      Washington   
210   Los Angeles County         6037           Texas   
1127  Los Angeles County         6037          Oregon   
246   Los Angeles County         6037        New York   
185   Los Angeles County         6037        New York   
265   Los Angeles County         6037        Missouri   
240   Los Angeles County         6037            Ohio   
164   Los Angeles County         6037        Michigan   
276   Los Angeles County         6037        Kentucky   
219   Los Angeles County         6037       Louisiana   
146   Los Angeles County         6037     Connecticut   
433   Los Angeles County         6037        Missouri   
122   Los Angeles County         6037         Georgia   
167   Los Angeles County         6037          Oregon   
248   Los Angeles County         6037        Michigan   
132   Los Angeles County         6037           Texas   
229   Los Angeles County         6037    Pennsylvania   

                  County2Name  County2FIPS  County1Population  MovedIn  \
329               Erie County        36029            9988370      136   
237            St. Louis city        29510            9988370      138   
305        New Hanover County        37129            9988370      138   
220          Worcester County        25027            9988370      139   
272           Plymouth County        25023            9988370      141   
264     Prince William County        51153            9988370      142   
461             Morris County        34027            9988370      143   
314             Summit County        39153            9988370      144   
253          Champaign County        17019            9988370      145   
533           Hamilton County        18057            9988370      153   
259            Pulaski County         5119            9988370      156   
344          Middlesex County        34023            9988370      162   
507            Fayette County        21067            9988370      166   
565           Okanogan County        53047            9988370      172   
210             Brazos County        48041            9988370      173   
1127              Lake County        41037            9988370      179   
246            Suffolk County        36103            9988370      181   
185           Onondaga County        36067            9988370      199   
265             Greene County        29077            9988370      199   
240           Hamilton County        39061            9988370      204   
164             Ingham County        26065            9988370      210   
276          Jefferson County        21111            9988370      211   
219   East Baton Rouge Parish        22033            9988370      220   
146         New London County         9011            9988370      228   
433            Jackson County        29095            9988370      247   
122            Chatham County        13051            9988370      247   
167            Jackson County        41029            9988370      262   
248               Kent County        26081            9988370      262   
132          Fort Bend County        48157            9988370      277   
229             Centre Co

In [363]:
len(comb_in[comb_in.MovedIn > 100])

99

In [364]:
len(comb_in[comb_in.MovedIn > 200])

57

In [362]:
len(comb_in[((comb_in.MovedOut > 100) & (comb_in.MovedIn > 100))])

61

In [365]:
len(comb_in[((comb_in.MovedIn - comb_in.MovedOut > 50) & (comb_in.MovedIn > 100))])

80

In [368]:
len(comb_in[(comb_in.MovedIn - comb_in.MovedOut > 50) & (comb_in.MovedOut > 100) & (comb_in.MovedIn > 100)])

46

In [ ]:
alt.Chart(comb_in).mark_circle().encode(x='', y='Med_Rent')

---

In [327]:
len(comb_out)

801

In [357]:
len(comb_out[comb_out.MovedOut - comb_out.MovedIn > 50])

223

In [358]:
len(comb_out[comb_out.MovedOut > 100])

184

In [359]:
len(comb_out[comb_out.MovedOut > 200])

94

In [351]:
len(comb_out[((comb_out.MovedOut > 100) & (comb_out.MovedIn > 100))])

80

In [355]:
len(comb_out[((comb_out.MovedOut - comb_out.MovedIn > 50) & (comb_out.MovedOut > 100))])

160

In [367]:
len(comb_out[(comb_out.MovedOut - comb_out.MovedIn > 50) & (comb_out.MovedOut > 100) & (comb_out.MovedIn > 100)])

66

In [314]:
comb_in.iloc[:5,:24]

County1Name  County1FIPS            State2Name  \
3   Los Angeles County         6037              New York   
6   Los Angeles County         6037              Illinois   
12  Los Angeles County         6037              New York   
13  Los Angeles County         6037                Hawaii   
16  Los Angeles County         6037  District of Columbia   

             County2Name  County2FIPS  County1Population  MovedIn  MovedOut  \
3        New York County        36061            9988370     3804      3316   
6            Cook County        17031            9988370     3784      2402   
12          Kings County        36047            9988370     3440      1298   
13       Honolulu County        15003            9988370     1855      1280   
16  District of Columbia        11001            9988370     1275      1175   

    Net_Dom  NonMovers  ...  To_Diff_County_Same_State  To_Diff_State  To_PR  \
3       488    8849567  ...                     180991         135437    370   
6      1382    8849567  ...                     180991         135437    370   
12     2142    8849567  ...                     180991         135437    370   
13      575    8849567  ...                     180991         135437    370   
16      100    8849567  ...                     180991         135437    370   

    Year                                     Address  MovedIn_Str  \
3   2017                   New York County, New York        3,804   
6   2017                       Cook County, Illinois        3,784   
12  2017                      Kings County, New York        3,440   
13  2017                     Honolulu County, Hawaii        1,855   
16  2017  District of Columbia, District of Columbia        1,275   

    MovedOut_Str  MovedNet_Str   FIPS Med_Mort  
3          3,316           488  36061   1889.0  
6          2,402         1,382  17031   1446.0  
12         1,298         2,142  36047   1844.0  
13         1,280           575  15003   1873.0  
16         1,175           100  11001   1955.0  

[5 rows x 24 columns]

In [315]:
comb_out.iloc[:5,:24]

County1Name  County1FIPS  State2Name      County2Name  County2FIPS  \
0  Los Angeles County         6037      Nevada     Clark County        32003   
1  Los Angeles County         6037     Arizona  Maricopa County         4013   
2  Los Angeles County         6037  Washington      King County        53033   
4  Los Angeles County         6037       Texas    Harris County        48201   
5  Los Angeles County         6037       Texas    Dallas County        48113   

   County1Population  MovedIn  MovedOut  Net_Dom  NonMovers  ...  \
0            9988370     4724     14191    -9467    8849567  ...   
1            9988370     3053      6494    -3441    8849567  ...   
2            9988370     2949      4147    -1198    8849567  ...   
4            9988370     2012      2830     -818    8849567  ...   
5            9988370     1354      2602    -1248    8849567  ...   

   To_Diff_County_Same_State  To_Diff_State  To_PR  Year  \
0                     180991         135437    370  2017   
1                     180991         135437    370  2017   
2                     180991         135437    370  2017   
4                     180991         135437    370  2017   
5                     180991         135437    370  2017   

                    Address  MovedIn_Str  MovedOut_Str  MovedNet_Str   FIPS  \
0      Clark County, Nevada        4,724        14,191        -9,467  32003   
1  Maricopa County, Arizona        3,053         6,494        -3,441   4013   
2   King County, Washington        2,949         4,147        -1,198  53033   
4      Harris County, Texas        2,012         2,830          -818  48201   
5      Dallas County, Texas        1,354         2,602        -1,248  48113   

  Med_Mort  
0   1159.0  
1   1144.0  
2   1878.0  
4   1191.0  
5   1161.0  

[5 rows x 24 columns]

In [316]:
comb_in['Med_Rent_Bin'] = comb_in.Med_Rent.round().values
comb_in['Med_Mort_Bin'] = comb_in.Med_Mort.round().values

C:\Users\Mahmoud\Miniconda3\envs\data_analysis\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.
C:\Users\Mahmoud\Miniconda3\envs\data_analysis\lib\site-packages\ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [317]:
comb_out['Med_Rent_Bin'] = comb_out.Med_Rent.round().values
comb_out['Med_Mort_Bin'] = comb_out.Med_Mort.round().values

C:\Users\Mahmoud\Miniconda3\envs\data_analysis\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.
C:\Users\Mahmoud\Miniconda3\envs\data_analysis\lib\site-packages\ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [318]:
comb_in.Med_Rent_Bin = [int(round(i/50)*50) for i in comb_in.Med_Rent_Bin]
comb_in.Med_Mort_Bin = [int(round(i/50)*50) for i in comb_in.Med_Mort_Bin]

In [319]:
comb_out.Med_Rent_Bin = [int(round(i/50)*50) for i in comb_out.Med_Rent_Bin]
comb_out.Med_Mort_Bin = [int(round(i/50)*50) for i in comb_out.Med_Mort_Bin]

In [320]:
rent_in_vals_dict = {}
for val in comb_in.Med_Rent_Bin.dropna().unique():
    bin_fips = comb_in.FIPS[comb_in.Med_Rent_Bin == val].values
    rent_in_vals_dict[val] = len(bin_fips)

In [321]:
rent_out_vals_dict = {}
for val in comb_out.Med_Rent_Bin.dropna().unique():
    bin_fips = comb_out.FIPS[comb_out.Med_Rent_Bin == val].values
    rent_out_vals_dict[val] = len(bin_fips)

In [335]:
mort_in_vals_dict = {}
for val in comb_in.Med_Mort_Bin.dropna().unique():
    bin_fips = comb_in.FIPS[comb_in.Med_Mort_Bin == val].values
    mort_in_vals_dict[val] = len(bin_fips)

In [336]:
mort_out_vals_dict = {}
for val in comb_out.Med_Mort_Bin.dropna().unique():
    bin_fips = comb_out.FIPS[comb_out.Med_Mort_Bin == val].values
    mort_out_vals_dict[val] = len(bin_fips)

In [322]:
rent_in_df = pd.DataFrame(rent_in_vals_dict.items())
rent_in_df.columns = ['Bin', 'Count']

In [323]:
rent_out_df = pd.DataFrame(rent_out_vals_dict.items())
rent_out_df.columns = ['Bin', 'Count']

In [337]:
mort_in_df = pd.DataFrame(mort_in_vals_dict.items())
mort_in_df.columns = ['Bin', 'Count']

In [338]:
mort_out_df = pd.DataFrame(mort_out_vals_dict.items())
mort_out_df.columns = ['Bin', 'Count']

#### *Layer charts to show difference (out behind in- since in counties are less)

In [339]:
alt.Chart(rent_in_df, title='Rent In').mark_bar(size=15.5).encode(x='Bin', y='Count')

alt.Chart(...)

In [340]:
alt.Chart(rent_out_df, title='Rent Out').mark_bar(size=15.5).encode(x='Bin', y='Count')

alt.Chart(...)

In [345]:
alt.Chart(mort_in_df, title='Mort In').mark_bar(size=11.5).encode(x='Bin', y='Count')

alt.Chart(...)

In [346]:
alt.Chart(mort_out_df, title='Mort Out').mark_bar(size=11.5).encode(x='Bin', y='Count')

alt.Chart(...)

---

### Findings:
- More counties where more people move out of LA (801) than in (432)
- Counties where more people move to LA have higher rents and morts (based on distribution)

In [165]:
ga = combined[(combined.County2FIPS > 13000) & (combined.County2FIPS < 14000)]
ga

County1Name  County1FIPS State2Name      County2Name  County2FIPS  \
6   Los Angeles County         6037    Georgia    DeKalb County        13089   
32  Los Angeles County         6037    Georgia    Fulton County        13121   
76  Los Angeles County         6037    Georgia      Cobb County        13067   
86  Los Angeles County         6037    Georgia  Gwinnett County        13135   

    County1Population  MovedIn  MovedOut  Net_Dom  NonMovers  ...  \
6             9988370      438       317      121    8849567  ...   
32            9988370      940       658      282    8849567  ...   
76            9988370      149       527      378    8849567  ...   
86            9988370      146       437      291    8849567  ...   

    MovedNet_Str     Type             More   FIPS  Med_Mort  Med_Rent  \
6            121   Net In        LA County  13089    1230.0    1062.0   
32           282   Net In        LA County  13121    1458.0    1089.0   
76          -378  Net Out      Cobb County  13067    1294.0    1102.0   
86          -291  Net Out  Gwinnett County  13135    1318.0    1142.0   

    Med_Mort_Inc  Med_Rent_Inc Mort_Share_Inc Rent_Share_Inc  
6         6480.0        3295.0           19.0           32.2  
32        8032.0        3314.0           18.2           32.9  
76        7658.0        3799.0           16.9           29.0  
86        6691.0        3493.0           19.7           32.7  

[4 rows x 31 columns]

---

In [129]:
out_ca.MovedIn.sum()

100729

In [73]:
out_ca.sort_values('MovedIn', ascending=False).head(50).MovedIn.sum()

56106

In [78]:
56106/100729

0.5569994738357374

---

In [75]:
out_ca.MovedOut.sum()

135807

In [76]:
out_ca.sort_values('MovedOut', ascending=False).head(50).MovedOut.sum()

74030

In [77]:
74030/135807

0.5451118130876906